libs/app-example-auth/src/lib/user-level.guard.ts

```diff
@@ -0,0 +1,30 @@
+import { CanMatchFn, Router } from '@angular/router'
+import { AuthStateService } from './auth-state.service'
+import { UserLevel } from '@myorg/app-example-models'
+import { inject } from '@angular/core'
+
+export const userLevelGuard: CanMatchFn = (route, segments) => {
```

Comment:
We need to provide a argument to the function so that we can tell which role can access. Eg-

```
export function userLevelGuardFn(allowedLevels: UserLevel[]): CanMatchFn {
    return (route, segments) => {
...
    }
}
```